doc: update doc for 2.5.2 (#13597)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-2.5.2.rst

.. _ncs_release_notes_252:

|NCS| v2.5.2 Release Notes
##########################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, nRF70, and nRF91 Series.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>` and :ref:`nRF Connect SDK v2.5.1 <ncs_release_notes_251>`.

> The changes affect Matter, WiFi, and SoftDevice controller.

Reviewer: ```suggestion
The changes affect Matter, Wi-Fi®, and SoftDevice Controller.
```

> The changes affect Matter, WiFi®, and SoftDevice Controller.

Reviewer 2: ```suggestion
The changes affect Matter, Wi-Fi®, and SoftDevice Controller.
```

Modem-related libraries and versions
====================================

.. list-table:: Modem-related libraries and versions
   :widths: 15 10
   :header-rows: 1

   * - Library name
     - Version information
   * - Modem library
     - `Changelog <Modem library changelog for v2.5.2_>`_
   * - LwM2M carrier library
     - `Changelog <LwM2M carrier library changelog for v2.5.2_>`_

Known issues
************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v2.5.2`_ for the list of issues valid for the latest release.

.. _ncs_release_notes_252_changelog:

Changelog
*********

The following sections provide detailed lists of changes by component.

Protocols
=========

This section provides detailed lists of changes by :ref:`protocol <protocols>`.

Matter
------

* Fixed:

  * An issue where the Matter device's application crashes if the DNS resolve response is processed with the TXT record's data size equal to ``0`` (KRKNWK-18256).
  * An issue with the IPv6 address update in the DNS server for Matter over Wi-Fi (KRKNWK-18256).
  * An issue with a memory leak in the deferred attribute persister (KRKNWK-18221).

Wi-Fi
-----

> * Fixed an issue with QSPI initialization on the custom board with nRF5340 and nRF7002 (SHEL-2372).

Reviewer 2: ```suggestion
* Fixed an issue with QSPI initialization on a custom board with nRF5340 and nRF7002 (SHEL-2372).
```

sdk-nrfxlib
-----------

> * Fixed some issues related to the :ref:`softdevice_controller` library.

Reviewer 2: ```suggestion
* Fixed issues related to the :ref:`softdevice_controller` library.
```

.. _ncs_release_notes_252:

|NCS| v2.5.2 Release Notes
##########################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, nRF70, and nRF91 Series.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>` and :ref:`nRF Connect SDK v2.5.1 <ncs_release_notes_251>`.

> [TODO]

Reviewer 3: ```suggestion
The changes affect Matter, WiFi, and SoftDevice controller. 
```

.. _ncs_release_notes_252:

|NCS| v2.5.2 Release Notes
##########################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, nRF70, and nRF91 Series.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>` and :ref:`nRF Connect SDK v2.5.1 <ncs_release_notes_251>`.

> [TODO]

Reviewer 4: Please add the following points under Matter section:

```
Matter
------

* Fixed:

  * An issue with processing the DNS resolve response with TXT record's data size equal to 0 (KRKNWK-18256).
  * An issue with IPv6 address update in DNS Server for Matter over Wi-Fi (KRKNWK-18256).
  * An issue with memory leak in deferred attribute persister (KRKNWK-18221).
```

> [TODO]

Reviewer 5: Please add below for Wi-Fi section:

```
Wi-Fi
-------
* Fixed an issue with QSPI initialisation on custom board with nRF7002 (SHEL-2372)
```

Reviewer 5: the commit for the fix has a typo, it points to `SHEL-2375`, not sure if we should mention it.

Author: > the commit for the fix has a typo, it points to `SHEL-2375`, not sure if we should mention it.

I would say this is not required. Not something the user needs to know.
